imod_coupler/drivers/dfm_metamod/dfm_metamod.py

```diff
@@ -148,38 +159,49 @@ def update(self) -> None:
         self.number_substeps_per_modflowstep = int(self.delt_mf6 / self.delt_msw_dflow)
         self.msw.prepare_time_step(self.delt_mf6)
 
-        # initialise water balance
-        self.exchange_balans.reset()
+        # initialise water balance 1d + 2d
+        self.exchange_balans_1d.reset()
+        self.exchange_balans_2d.reset()
 
         # stage from dflow 1d to modflow active coupled riv
         self.exchange_stage_1d_dfm2mf6()
 
-        # flux from modflow active coupled riv to water balance
+        # flux from modflow active coupled riv to water balance 1d
         self.exchange_flux_riv_active_mf62dfm()
 
-        # flux from modflow passive coupled riv to water balance
+        # flux from modflow passive coupled riv to water balance 1d
         self.exchange_flux_riv_passive_mf62dfm()
 
-        # flux from modflow passive coupled drn to water balance
+        # flux from modflow passive coupled drn to water balance 1d
         self.exchange_flux_drn_passive_mf62dfm()
 
         # sub timestepping between metaswap and dflow
         subtimestep_endtime = t_begin
         for idtsw in range(self.number_substeps_per_modflowstep):
             subtimestep_endtime += self.delt_msw_dflow
+
+            # initial 2d stage from dflow 2d to msw
+            self.exchange_stage_2d_dfm2msw()
+
+            # initiate surface water timestep
             self.msw.start_surface_water_time_step(idtsw)
-            # flux from metaswap ponding to water balance
+
+            # flux from metaswap ponding to water balance 1d
             self.exchange_ponding_msw2dflow1d()
 
-            # flux from metaswap sprinkling to water balance
+            # flux from metaswap sprinkling to water balance 1d
             self.exchange_sprinkling_msw2dflow1d()
 
-            # exchange water balance to dlfow
-            self.exchange_balans.sum_demand()
-            self.exchange_balans_2dflow(self.exchange_balans.demand["sum"])
+            # exchange ponding msw to water balance 2d
+            self.exchange_ponding_msw2dflow2d()
+
+            # exchange water balance 1d to dlfow 1d
+            self.exchange_balans_1d.sum_demand()
+            self.exchange_balans_2dfm(self.exchange_balans_1d.demand["sum"])
 
             # get cummelative flux before dfm-run
-            q_dflow0 = self.dfm.get_cumulative_fluxes_1d_nodes()
+            q_dflow0_1d = self.dfm.get_cumulative_fluxes_1d_nodes()
```

Comment:
feel free to use longer function names. maybe q_dflow_1d_before_solve or something like that

Reply:
done